src/x11/screen.c

```diff
@@ -82,27 +82,33 @@ void alloc_screen_ar(int n)
 
 int randr_event_base = 0;
 
+static int randr_major_version = 1;
+static int randr_minor_version = 5;
+
 void randr_init()
 {
         int randr_error_base = 0;
         if (!XRRQueryExtension(xctx.dpy, &randr_event_base, &randr_error_base)) {
                 fprintf(stderr, "Could not initialize the RandR extension, falling back to single monitor mode.\n");
                 return;
         }
+        XRRQueryVersion(xctx.dpy, &randr_major_version, &randr_minor_version);
         XRRSelectInput(xctx.dpy, RootWindow(xctx.dpy, DefaultScreen(xctx.dpy)), RRScreenChangeNotifyMask);
 }
 
 void randr_update()
 {
-        int n;
-        XRRMonitorInfo *m = XRRGetMonitors(xctx.dpy, RootWindow(xctx.dpy, DefaultScreen(xctx.dpy)), true, &n);
-
-        if (m == NULL || n == -1) {
+        if (randr_minor_version < 5) {
```

Comment:
You're only checking for `minor < 5` here which is going to break at the next major version. A more complete check would be `randr_major_version > 1 || (randr_major_version == 1 && randr_minor_version < 5)`

Additionally, there should probably be a more descriptive error message mentioning the outdated version.

Reply:
Good point.